share/functions/__fish_git_prompt.fish

```diff
@@ -417,6 +417,14 @@ function __fish_git_prompt --description "Prompt function for Git"
         set u "$___fish_git_prompt_color_untrackedfiles$u$___fish_git_prompt_color_untrackedfiles_done"
     end
     set b (string replace refs/heads/ '' -- $b)
+     if test -z "$__fish_git_prompt_branch_max_length"
+        set __fish_git_prompt_branch_max_length 20
+    end
+    set -l b_ori_length (string length $b)
+    set b (string sub -l $__fish_git_prompt_branch_max_length $b)
+    if test $b_ori_length -gt $__fish_git_prompt_branch_max_length
+        set b "$b…"
```

Comment:
It would probably be nice to have a `string shorten` subcommand that does all this. It could then incorporate some more logic, e.g. using `…` only if the current locale/encoding allows it.

And fish internally is _already_ doing that - see `ellipsis_char` in src/common.cpp.